content/blog/write-git-commit-messages-first.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "How — and Why — to Write Commit Messages First"
+featured: false
+description: >
+  Writing your commit messages first keeps you and your pair focused between the feature you are currently trying to implement and whatever current change you are making that could be a smaller, more atomic pushable commit.
+date: 2020-08-19
+tags:
+  - Git
+# Author(s)
+team:
+  - Eric Tsiliacos
+---
+
+You’re halfway through delivering your feature and you decide to take a look at your diff. Doing so gives you a sinking feeling in your stomach, because you see a lot more changes than you were expecting, some of which were refactorings, like renames or structural changes you wish were separated into their own commits. Teasing apart these smaller commits can be messy and would take too much effort at this point. But if you’d tried to preemptively break them apart at random, you’d have run the risk of overengineering your work or creating unnecessary changes.
+
+Writing commit messages first can help pairs navigate between feature delivery and what could be smaller, more atomic commits. It’s analogous to working from “stories” off a backlog or trying to get the next failing test to pass. If you or your pair gets lost in the weeds, pointing back to the story provides a direct and egoless way to get back on track. In test-driven development, pairs stay focused by thinking about small, incremental, falsifiable functionality one failing test at a time. Writing commit messages first can help pairs articulate the space between a feature story and individual tests. 
```

Comment:
I'm not positive I'm being grammatically correct here, but for some reason this line reads a bit awkward to me. If the extra comma isn't the solution, might be a slight different way to phrase it?

```suggestion
Writing commit messages first can help pairs navigate between feature delivery and what could be smaller, more atomic commits. It’s analogous to working from “stories” off a backlog or trying to get the next failing test to pass. If you or your pair gets lost in the weeds, pointing back to the story provides a direct and egoless way to get back on track. In test-driven development, pairs stay focused by thinking about small, incremental, falsifiable functionality, one failing test at a time. Writing commit messages first can help pairs articulate the space between a feature story and individual tests. 
```

Reply:
I'm not a comma master, but it's probably okay either way. I'll leave it in deference to the copyeditor in this case.